snippets/standard/datetime/json/datetime-converter-examples/example1/Program.cs

```diff
@@ -0,0 +1,78 @@
+﻿using System;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+using System.Text.RegularExpressions;
+
+namespace DateTimeConverterExamples
+{
+    public class DateTimeConverterExample1 : JsonConverter<DateTime>
+    {
+        public override DateTime Read(ref Utf8JsonReader reader, Type typeToConvert, JsonSerializerOptions options)
```

Comment:
Should there be a validation check that `typeToConvert == typeof(DateTime)`?

In this snippet:
https://github.com/dotnet/samples/blob/ee556965f4fa50cda130816dc5ad75714bdb7b4d/snippets/standard/datetime/json/datetime-converter-examples/example1/Program.cs#L10-L18

@JeremyKuhne, @steveharter - maybe you can help answer this

Reply:
No you don't need a check on `typeToConvert`. It will always be `typeof(DateTime)`.

The `typeToConvert` parameter is useful for polymorphic cases and when using generics to get `typeof(<T>)` value in a performant way.